components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java

```diff
@@ -0,0 +1,617 @@
+                                ", User ID : " + authorizedUser + " and Scope : " + scope);
+                    }
+                }
+            }
+
+            // check if the last issued access token is still active and valid in the database
+            AccessTokenDO existingAccessTokenDO = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                    .getLatestAccessToken(consumerKey, authorizationReqDTO.getUser(), userStoreDomain, scope, false);
+
+            if (existingAccessTokenDO != null) {
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Retrieved latest Access Token" +
+                            " for Client ID : " + consumerKey + ", User ID :" + authorizedUser +
+                            " and Scope : " + scope + " from database");
+                }
+
+                long expiryTime = OAuth2Util.getTokenExpireTimeMillis(existingAccessTokenDO);
+
+                long refreshTokenExpiryTime = OAuth2Util.getRefreshTokenExpireTimeMillis(existingAccessTokenDO);
+
+                if (OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE.equals(
+                        existingAccessTokenDO.getTokenState()) && (expiryTime > 0 || expiryTime < 0)) {
+
+                    // token is active and valid
+                    if (log.isDebugEnabled()) {
+                        if (expiryTime > 0) {
+                            log.debug("Access token is valid for another " + expiryTime + "ms");
+                        } else {
+                            log.debug("Infinite lifetime Access Token found in cache");
+                        }
+                    }
+
+                    if (cacheEnabled) {
+                        OAuthCache.getInstance().addToCache(cacheKey, existingAccessTokenDO);
+                        if (log.isDebugEnabled()) {
+                            log.debug("Access Token was added to cache for cache key : "
+                                    + cacheKey.getCacheKeyString());
+                        }
+                    }
+                    return existingAccessTokenDO;
+
+                } else {
+
+                    if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable
+                            (IdentityConstants.IdentityTokens.ACCESS_TOKEN)) {
+                        log.debug("Access Token is " + existingAccessTokenDO.getTokenState());
+                    }
+                    String tokenState = existingAccessTokenDO.getTokenState();
+
+                    if (OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE.equals(tokenState)) {
+
+                        // Token is expired. If refresh token is still valid, use it.
+                        if (refreshTokenExpiryTime > 0 || refreshTokenExpiryTime < 0) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Access token has expired, But refresh token is still valid. User existing " +
```

Comment:
Change the string from "user" to "use"

Reply:
Fixed - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1226